src/wp-includes/taxonomy.php

```diff
@@ -4243,226 +4225,15 @@ function _update_generic_term_count( $terms, $taxonomy ) {
-	}
-
-	$children_tt_ids = $wpdb->get_col( $wpdb->prepare( "SELECT term_taxonomy_id FROM $wpdb->term_taxonomy WHERE parent = %d AND taxonomy = %s", $term_id, $term_taxonomy->taxonomy ) );
-	if ( ! empty( $children_tt_ids ) ) {
-		foreach ( $children_tt_ids as $child_tt_id ) {
-			$wpdb->update(
-				$wpdb->term_taxonomy,
-				array( 'parent' => $new_term_id ),
-				array( 'term_taxonomy_id' => $child_tt_id )
-			);
-			clean_term_cache( (int) $child_tt_id, '', false );
-		}
-	} else {
-		// If the term has no children, we must force its taxonomy cache to be rebuilt separately.
-		clean_term_cache( $new_term_id, $term_taxonomy->taxonomy, false );
-	}
-
-	clean_term_cache( $term_id, $term_taxonomy->taxonomy, false );
-
-	/*
-	 * Taxonomy cache clearing is delayed to avoid race conditions that may occur when
-	 * regenerating the taxonomy's hierarchy tree.
-	 */
-	$taxonomies_to_clean = array( $term_taxonomy->taxonomy );
-
-	// Clean the cache for term taxonomies formerly shared with the current term.
-	$shared_term_taxonomies = $wpdb->get_col( $wpdb->prepare( "SELECT taxonomy FROM $wpdb->term_taxonomy WHERE term_id = %d", $term_id ) );
-	$taxonomies_to_clean    = array_merge( $taxonomies_to_clean, $shared_term_taxonomies );
-
-	foreach ( $taxonomies_to_clean as $taxonomy_to_clean ) {
-		clean_taxonomy_cache( $taxonomy_to_clean );
-	}
-
-	// Keep a record of term_ids that have been split, keyed by old term_id. See wp_get_split_term().
-	if ( $record ) {
-		$split_term_data = get_option( '_split_terms', array() );
-		if ( ! isset( $split_term_data[ $term_id ] ) ) {
-			$split_term_data[ $term_id ] = array();
-		}
-
-		$split_term_data[ $term_id ][ $term_taxonomy->taxonomy ] = $new_term_id;
-		update_option( '_split_terms', $split_term_data );
-	}
-
-	// If we've just split the final shared term, set the "finished" flag.
-	$shared_terms_exist = $wpdb->get_results(
-		"SELECT tt.term_id, t.*, count(*) as term_tt_count FROM {$wpdb->term_taxonomy} tt
-		 LEFT JOIN {$wpdb->terms} t ON t.term_id = tt.term_id
-		 GROUP BY t.term_id
-		 HAVING term_tt_count > 1
-		 LIMIT 1"
-	);
-	if ( ! $shared_terms_exist ) {
-		update_option( 'finished_splitting_shared_terms', true );
-	}
-
-	/**
-	 * Fires after a previously shared taxonomy term is split into two separate terms.
-	 *
-	 * @since 4.2.0
-	 *
-	 * @param int    $term_id          ID of the formerly shared term.
-	 * @param int    $new_term_id      ID of the new term created for the $term_taxonomy_id.
-	 * @param int    $term_taxonomy_id ID for the term_taxonomy row affected by the split.
-	 * @param string $taxonomy         Taxonomy for the split term.
-	 */
-	do_action( 'split_shared_term', $term_id, $new_term_id, $term_taxonomy_id, $term_taxonomy->taxonomy );
-
-	return $new_term_id;
+	return $term_id;
 }
 
 /**
  * Splits a batch of shared taxonomy terms.
  *
  * @since 4.3.0
- *
- * @global wpdb $wpdb WordPress database abstraction object.
  */
 function _wp_batch_split_terms() {
```

Comment:
Why keep these functions around as empty? (This feedback applies to all similarly treated functions in this file.)

If this is about backward compatibility, I'd argue it's not worth it here for a few reasons:
* This PR already breaks backward compatibility and (rightfully) prevents updates in case the site still needs the functionality.
* These functions are marked private, so they should never be used externally in the first place.

At the very minimum, if we keep them, I think they should be marked deprecated and moved to `deprecated.php`. But per the above, I think it's really not worth keeping them. We should have a dev note for this change, and there we could also mention that all the private helper functions are being removed.